### ISSUE TITLE
Raiden node unable to process payments after a while

### ISSUE THREAD

**Issue by CosminNechifor** [raiden-network/raiden#2293]
## Problem Definition

I have tried a second time to get my node stuck, and make it unable to process payments. 
Steps I've done to get it stuck: 

1. I have created a new geth account, transfered eth to it and minted tokens. After that I restarted geth.
2. I started the raiden client (0.6.0 release, same thing as @czepluch )
3. I tried to send Jacob a payment and it worked
4. I run the following [script](https://gist.github.com/CosminNechifor/896fc011c0f6c3d21f5da2204920df73) to send 50 payments of 1 token to Jacob. The first time, all payments went through and he received them. 
5. When I started it a second time (I wanted to send him 200 more payments of 1 token), it got stuck after I think 20 transactions, then I [canceled](https://gist.github.com/CosminNechifor/6dde3f58e3a1395c4ba020c56d325b64) it.

Even after restarting the node, I could no longer perform payments to him. 

### Logs:

[Jacob](https://gist.github.com/czepluch/984d466764cb12d16905bf0ff5069516)
[Cosmin](https://gist.github.com/CosminNechifor/26a9e913a9c803f86fe3c9d6d90bf2cc)

If anyone else has the time, I strongly recommend trying to see if this would happen to their node also.   

### System Description

```
Linux cosmin-Lenovo-ideapad-100-15IBD 4.15.0-33-generic #36-Ubuntu SMP Wed Aug 15 16:00:05 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
/usr/bin/solc
solc, the solidity compiler commandline interface
Version: 0.4.24+commit.e67f0147.Linux.g++
/usr/bin/geth
Geth
Version: 1.8.12-stable
Git Commit: 37685930d953bcbe023f9bc65b135a8d8b8f1488
Architecture: amd64
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.10.1
Operating System: linux
GOPATH=
GOROOT=/usr/lib/go-1.10
```

**Comment by hackaugusto** [comment]
> Please write a description of what we discussed in-person is the side-effect that causes this bug and don't forget to add a changelog entry with your fix

This bug was caused because the arguments `message_identifier` and `payment_identifier` where swapped.

This call:

https://github.com/raiden-network/raiden/blob/b883acd1ab28187c3bba04e7161b62fa1b83ca87/raiden/transfer/mediated_transfer/initiator.py#L251-L257

Must have this order:

https://github.com/raiden-network/raiden/blob/b883acd1ab28187c3bba04e7161b62fa1b83ca87/raiden/transfer/channel.py#L1301-L1307

**Comment by CosminNechifor** [comment]
Fixed by: #2299
